### DISCUSSION TITLE
No totalCount for pagination?

### DISCUSSION THREAD

**Discussion by ghost** [Azure/data-api-builder#2234] [Q&A] [answered]
My only exposure to DAB so far is through the SWA data-api. This is already a super impressive tool and I really hope it can go very far!

My question is to do with pagination and totals.

I see there's pagination support with the 'next page' link - but I expected there to be a way to get a 'totalCount' - especially in the GraphQL endpoint. Am I missing a way to do this already? Or is this already on a backlog?

I can see something to do with OData $count in the roadmap - which will need the value I'm looking for - but adding it _only_ for OData support seems strange, when it looks to be a fairly normal thing to have in a GraphQL query?

**Comment by JerryNixon** [comment, marked as answer]
I assume you have discovered that you can do this in two calls where the first gets the values the second gets the counts? I also assume you have considered that you can do this in a single call if you have a view with an additional column? I am not saying this is the best approach, just that these are workarounds that others are doing. We have enhancing pagination on our roadmap.
